file-formats/sush/type/zif_aff_sush_v1.intf.abap

```diff
@@ -0,0 +1,186 @@
+      "! Application Is Obsolete
+      o_flag type ty_usob_status_flag value 'O',
+    end of co_usob_status_flag,
+
+    "! <p class="shorttext">Maintenance Status</p>
+    "! Maintenance Status
+    begin of co_okflag,
+      "! <p class="shorttext">No Default</p>
+      "! No Default
+      x_flag type ty_okflag value 'X',
+      "! <p class="shorttext">Default With Field Values</p>
+      "! Default With Field Values
+      y_flag type ty_okflag value 'Y',
+      "! <p class="shorttext">Default Without Field Values</p>
+      "! Default Without Field Values
+      v_flag type ty_okflag value 'V',
+      "! <p class="shorttext">Default Inactive</p>
+      "! Default Inactive
+      i_flag type ty_okflag value 'I',
+      "! <p class="shorttext">Unmaintained Object Status</p>
+      "! Unmaintained Object Status
+      u_flag type ty_okflag value 'U',
+      "! <p class="shorttext">Authorization Check Inactive</p>
+      "! Authorization Check Inactive
+      n_flag type ty_okflag value 'N',
+    end of co_okflag.
+
+  types:
+    "! <p class="shorttext">Application Name</p>
+    "! Name of Application
+    ty_name type c length 30,
+    "! <p class="shorttext">Application Type</p>
+    "! Type of Application
+    ty_type type c length 2.
+
+  types:
+    "! <p class="shorttext">General</p>
+    "! General
+    "! $required
+    begin of ty_general,
+      "! <p class="shorttext">Application Name</p>
+      "! Application Name of Authorization Default Value.
+      "! $required
+      name   type ty_name,
+      "! <p class="shorttext">Type</p>
+      "! Type of Authorization Default Value.
+      "! $required
+      type   type ty_type,
+      "! <p class="shorttext">Maintenance Mode</p>
+      "! Maintenance Mode.
+      "! $values {@link zif_aff_sush_v1.data:co_usob_status_flag}
+      "! $required
+      okflag type ty_usob_status_flag,
+    end of ty_general .
+
+  types:
+    "! <p class="shorttext">Authorization Value</p>
+    "! Authorization Value
+    ty_val type c length 40.
+
+  types:
+    "! <p class="shorttext">Default Authorization Values</p>
+    "! Default Authorization Values
+    begin of ty_value,
+      "! <p class="shorttext">From</p>
+      "! From Value
+      "! $showAlways
+      low  type ty_val,
+      "! <p class="shorttext">To</p>
+      "! To Values
+      "! $showAlways
+      high type ty_val,
+    end of ty_value .
+
+  types:
+    "! <p class="shorttext">Authorization Fields</p>
+    "! Authorization Fields
+    ty_values  type standard table of ty_value with default key.
+
+  types:
+    "! <p class="shorttext">Authorization Field</p>
+    "! Authorization Field
+    ty_field type c length 10.
+
+  types:
+    "! <p class="shorttext">Authorization Field Details</p>
+    "! Authorization Field Details
+    begin of ty_fldtablestruc,
+      "! <p class="shorttext">Field</p>
+      "! Name of Authorization Field
+      "! $required
+      field  type ty_field,
```

Comment:
Also here, you might want to consider whether we should just use `name`. Any other opinions?